src/main/java/core/basesyntax/dao/FruitFileReader.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.exceptions.InvalidLineException;
+import core.basesyntax.exceptions.NullFileNameException;
+import core.basesyntax.service.impl.FruitTransaction;
+import core.basesyntax.service.impl.OperationHandler;
+import core.basesyntax.service.impl.OperationStrategyImpl;
+import core.basesyntax.strategy.OperationStrategy;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class FruitFileReader implements FruitDataReader {
+    private static final int NECESSARY_LINE_ELEMENT_NUMBER = 3;
+    private static final String EMPTY_LINE = "";
+    private final String fileName;
+    private final OperationStrategy operationStrategy;
+
+    public FruitFileReader(String fileName) {
+        if (fileName == null) {
+            throw new NullFileNameException("Can't get a file with NULL name");
+        }
+        this.fileName = fileName;
+        operationStrategy = new OperationStrategyImpl(new OperationHandler().getOperations());
+    }
+
+    @Override
+    public void read() {
```

Comment:
same here
reader should better only read
can we return List of Strings from here? and do all further processing in a different service TransactionService?